pages/listings.tsx

```diff
@@ -1,7 +1,70 @@
-export default function ListingsPage() {
+import AuthGuard from "@App/lib/auth/AuthGuard";
+import useAuthContext from "@App/lib/auth/AuthContext";
+import fetchcurrentuserListings from "@App/lib/listings/useFetchCurrentUserListing";
+import { Card, CardMedia, Stack, CardContent, Box, LinearProgress, Paper, Typography, Button } from "@mui/material";
+import { updateCurrentUser } from "firebase/auth";
+import type { NextPage } from "next";
+import {
+  IBaseListing,
+  IListingAttributes,
+  TListingCollectionReference,
+  TListingDocumentReference,
+} from "@App/lib/listings/types"
+
+const ListingsPage: NextPage = () => {
+ 
+  const {currentUser} = useAuthContext()
+  const { data: listings, isLoading } = fetchcurrentuserListings(  { userId: currentUser?.id }  );
+  if (isLoading || !listings) return <LinearProgress />;
+
+  if (listings.empty) return <Typography>No listings found</Typography>;
+
   return (
-    <div>
-      <h1>Listings</h1>
-    </div>
+    <AuthGuard>
+      <Typography variant="h2">My Books</Typography>
```

Comment:
We want to say `My Listings` instead here, because these are books the user wants to sell. There is a another page for the books that the user currently owns

Reply:
done